crud/src/main/java/com/redhat/lightblue/crud/DocumentStream.java

```diff
@@ -0,0 +1,77 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.crud;
+
+import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.function.Function;
+import java.util.function.Consumer;
+
+/**
+ * This interface is used to stream documents. A call to getDocuments
+ * should return an iterator of the document resultset.
+ */
+public interface DocumentStream<T> extends Iterator<T>{
+    /**
+     * Close the document stream
+     */
+    void close();
+
+    /**
+     * Adds a listener that will be called for each document of the stream when next() is called
+     */
+    void forEach(Consumer<T> dest);
```

Comment:
> But this isn't exactly a listener. It is a function that will be called for each doc, as they are pulled from the stream, so forEach tells what it does

Maybe I need to see an example of the method being used? I don't see any calls in this PR, and the only implementations I see just add the Consumer to an array called listeners. I don't even think I see a place that any action is being performed on those arrays.